dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/oidc/DhisClientRegistrationRepository.java

```diff
@@ -0,0 +1,94 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import org.hisp.dhis.external.conf.DhisConfigurationProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.core.AuthenticationMethod;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.security.oauth2.core.oidc.IdTokenClaimNames;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+
+import static org.hisp.dhis.external.conf.ConfigurationKey.OIDC_PROVIDER_CLIENT_ID;
+import static org.hisp.dhis.external.conf.ConfigurationKey.OIDC_PROVIDER_CLIENT_SECRET;
+import static org.hisp.dhis.external.conf.ConfigurationKey.OIDC_PROVIDER_REDIR_BASE_URL;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@Component( "dhisClientRegistrationRepository" )
+public class DhisClientRegistrationRepository
+    implements ClientRegistrationRepository
+{
+    private InMemoryClientRegistrationRepository repository;
+
+    @Autowired
+    public DhisConfigurationProvider dhisConfigurationProvider;
+
+    @PostConstruct
+    public void init()
+    {
+        String id = dhisConfigurationProvider.getProperty( OIDC_PROVIDER_CLIENT_ID );
+        String secret = dhisConfigurationProvider.getProperty( OIDC_PROVIDER_CLIENT_SECRET );
+        String redirBaseUri = dhisConfigurationProvider.getProperty( OIDC_PROVIDER_REDIR_BASE_URL );
+
+        //.redirectUriTemplate( "{baseUrl}/login/oauth2/code/{registrationId}" )
+        ClientRegistration idporten = ClientRegistration.withRegistrationId( "idporten" )
+            .clientId( id )
+            .clientSecret( secret )
+            .clientAuthenticationMethod( ClientAuthenticationMethod.BASIC )
+            .authorizationGrantType( AuthorizationGrantType.AUTHORIZATION_CODE )
+            .redirectUriTemplate( redirBaseUri + "/login/oauth2/code/{registrationId}" )
+            .scope( "openid", "profile" )
+            .authorizationUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/authorize" )
+            .tokenUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/token" )
+            .userInfoUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/userinfo" )
+            .userNameAttributeName( IdTokenClaimNames.SUB )
+            .userInfoAuthenticationMethod( AuthenticationMethod.HEADER )
+            .jwkSetUri( "https://oidc-ver2.difi.no/idporten-oidc-provider/jwk" )
+            .clientName( "Google" )
```

Comment:
Is "Google" correct here?

Reply:
Nope, det er glipp, ser ut til å fungere likevel